missingMapper function

aminya opened the pull request:
- Adds missingMapper function to map the missing files like mapping Release into Debug

Fixes #3 

Should be merged after #5

aminya commented:
@danielgindi Could you please review this?

aminya committed:
add replacements option

aminya committed:
add replacements to readme

danielgindi commented on README.md:
```diff
@@ -41,11 +41,15 @@ export default {
+            // OR you can have a function that returns a desired file name and a specific destination to copy to
             map: modulePath => { name: 'filename.node', copyTo: 'C:\\Dist\\libs\\filename.node' },
+
+            // an object that gives the replacement mappin in case the node module was missing
```
`an object that gives the replacement mapping in case of a missing node module`

aminya replied:
Fixed in https://github.com/danielgindi/rollup-plugin-natives/pull/4/commits/15de9d68a65598b505d201770e8d82003c34aea1

danielgindi commented on README.md:
```diff
@@ -41,11 +41,15 @@ export default {
+
+            // an object that gives the replacement mappin in case the node module was missing
+            // If not set the following is used. e.g. `Debug` is matched and replaced with `Release` via regex .
+            replacements: {'Debug': 'Release', 'Release':'Debug'}
```
Maybe make this a function? And a more meaningful name?
If it's a dictionary, it should be something like `missingMap`. But a function will be a better choice anyway as you are stepping into the realm of more complex configurations. So maybe a `function mapper(name: string, exists: boolean)`.

danielgindi commented on src/index.js:
```diff
@@ -181,9 +180,24 @@ function nativePlugin(options) {
-
                 if (isNew) {
-                    Fs.copyFileSync(nativePath, mapping.copyTo);
+                    if (Fs.pathExistsSync(nativePath))
```
So maybe here do a preliminary step - check for existence, and then go through the mapping function.  
If the function does not exist, the path does not change. If there is a mapper and the  result if `undefined` or `null` - also leave it unchanged.  

Finally, try to copy the file. Failure will emit a warning.

In the middle - if there's some complex logic to perform to provide the "replacement", the mapper function will do it. Including checking for existence of replacements etc.

This will space for almost any quirk anyone will want to use there.

aminya committed:
fix readme description

https://github.com/danielgindi/rollup-plugin-natives/pull/4#discussion_r453356405

danielgindi commented:
@aminya Could you please treat the reviewed issues and rebase on master?

danielgindi commented:
@aminya Is this pr alive?

aminya commented:
What else do you want me to do? I think I addressed the issues

danielgindi commented:
> What else do you want me to do? I think I addressed the issues

I left comments 3 years ago

danielgindi commented:
Implemented as a different feature. Closing.
